packages/tab-bar/index.js

```diff
@@ -0,0 +1,146 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+
+import TabScroller from '@material/react-tab-scroller';
+import {MDCTabBarFoundation} from '@material/tab-bar/dist/mdc.tabBar';
+
+export default class TabBar extends Component {
+  tabBarElement_ = React.createRef();
+  tabScroller_ = React.createRef();
+  tabList_ = [];
+
+  foundation_ = null;
+
+  componentDidMount() {
+    this.foundation_ = new MDCTabBarFoundation(this.adapter);
+    this.foundation_.init();
+    this.activateTab(this.props.activeIndex);
+  }
+
+  componentWillUnmount() {
+    this.foundation_.destroy();
+  }
+
+  get classes() {
+    return classnames('mdc-tab-bar', this.props.className);
+  }
+
+  get adapter() {
+    return {
+      scrollTo: (scrollX) => this.tabScroller_.current.scrollTo(scrollX),
+      incrementScroll: (scrollXIncrement) => this.tabScroller_.current.incrementScroll(scrollXIncrement),
+      getScrollPosition: () => this.tabScroller_.current.getScrollPosition(),
+      getScrollContentWidth: () => this.tabScroller_.current.getScrollContentWidth(),
+      getOffsetWidth: () => this.tabBarElement_.current.offsetWidth,
+      isRTL: () => !!this.props.isRTL,
+      activateTabAtIndex: (index, clientRect) => this.tabList_[index].activate(clientRect),
+      deactivateTabAtIndex: (index) => this.tabList_[index].deactivate(),
+      focusTabAtIndex: (index) => this.tabList_[index].focus(),
+      getTabIndicatorClientRectAtIndex: (index) => this.tabList_[index].computeIndicatorClientRect(),
+      getTabDimensionsAtIndex: (index) => this.tabList_[index].computeDimensions(),
+      getActiveTabIndex: () => this.props.activeIndex,
+      getFocusedTabIndex: () => {
+        const activeElement = document.activeElement;
+        return this.tabList_.indexOf((tab) => tab.tabElement_.current === activeElement);
+      },
+      getIndexOfTab: (tabToFind) => this.tabList_.indexOf(tabToFind),
+      getTabListLength: () => this.tabList_.length,
+      notifyTabActivated: (index) => this.props.handleActiveIndexUpdate(index),
+    };
+  }
+
+  activateTab(index) {
+    this.foundation_.activateTab(index);
+  }
+
+  scrollIntoView(index) {
```

Comment:
Also what if this was in the componentDidUpdate() method as well? And we created a new prop, like `props.tabToScroll`?

I just say that since its the user will need a `ref` to the tabbar in order to scrollIntoView(). I think we should minimize the amount we force the user to use a ref.